hamilton/plugins/xgboost_extensions.py

```diff
@@ -17,8 +17,8 @@
 from hamilton.io import utils
 from hamilton.io.data_adapters import DataLoader, DataSaver
 
-XGBOOST_MODEL_TYPES = [xgboost.XGBModel, xgboost.Booster]
-XGBOOST_MODEL_TYPES_ANNOTATION = Union[xgboost.XGBModel, xgboost.Booster]
+XGBOOST_MODEL_TYPES = [xgboost.XGBModel, xgboost.Booster, xgboost.XGBClassifier]
+XGBOOST_MODEL_TYPES_ANNOTATION = Union[xgboost.XGBModel, xgboost.Booster, xgboost.XGBClassifier]
```

Comment:
XGBoost documentation is hard to navigate but here are all the classes:
https://xgboost.readthedocs.io/en/stable/python/index.html

- xgboost.Booster is the core object; it is never used directly. It should be supported for users that don't want any other dependencies
- xgboost.XGBModel is the basis of the sklearn API (what most people use)
- xgboost.[DaskScikitLearnBase ](https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/dask/__init__.py#L1608)is the basis of the dask API. It subclasses xgboost.XGBModel
- _SparkXGBEstimator is the basis of the spark API, which is experimental. It subclasses pyspark.ml classes. It has dedicated reader and writers to interact with Booster objects [ref](https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/spark/core.py#L1542)